File: razer-sdk/src/main/java/com/bonepl/razersdk/animation/Frame.java
package com.bonepl.razersdk.animation;

import com.bonepl.razersdk.color.Color;
import com.bonepl.razersdk.sdk.RzKey;

import java.util.Collections;
import java.util.EnumMap;
import java.util.Map;

/**
 * Frame object containing map of keys to corresponding colors
 */
public class Frame {
    private final Map<RzKey, Color> keysToColors = new EnumMap<>(RzKey.class);

    /**
     * Create a frame from provided keys to colors map
     *
     * @param keysToColors map of keys to colors
     */
    public Frame(Map<RzKey, Color> keysToColors) {
        this.keysToColors.putAll(keysToColors);
    }

    /**
     * Return the underlying map of keys to colors
     *
     * @return map of keys to colors
     */
    public Map<RzKey, Color> getKeysToColors() {
        return Collections.unmodifiableMap(keysToColors);
    }
}

File: src/razer-sdk/animation/Frame.ts
import {Color} from '../color/Color.js';
import {RzKey} from '../sdk/RzKey.js';

/**
 * Frame object containing map of keys to corresponding colors
 */
export class Frame {
    private readonly keysToColors: Map<RzKey, Color>;

    constructor(keysToColors: Map<RzKey, Color>) {
        this.keysToColors = new Map(keysToColors);
    }

    /** Return the underlying map of keys to colors (as a copy) */
    getKeysToColors(): ReadonlyMap<RzKey, Color> {
        return this.keysToColors;
    }
}
